src/reporter.ts

```diff
@@ -0,0 +1,168 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import type {
+  JUnitTestSuite,
+  JUnitTestCase,
+  Failure,
+  Error,
+  Property,
+  ConfigOption,
+} from './type';
+import { create } from 'xmlbuilder2';
+
+export class TestSuite implements JUnitTestSuite {
+  name: string;
+  tests: number;
+  failures: number;
+  errors: number;
+  time: number;
+  timestamp: string;
+  file: string;
+  testSuites: JUnitTestSuite[];
+  testCases: JUnitTestCase[];
+  properties?: Property[] | undefined;
```

Comment:
Just curious why some properties are marked optional *and* have an explicit type union with `undefined`.

Reply:
hmm.. Cleaned it up.